test/test_stateszoo_api.jl

```diff
@@ -1,15 +1,44 @@
 using QuantumSavory
-using QuantumSavory.StatesZoo: ZALMSpinPair, SingleRailMidSwapBell, DualRailMidSwapBell
+using QuantumSavory.StatesZoo: ZALMSpinPairU, ZALMSpinPairN, SingleRailMidSwapBellU, SingleRailMidSwapBellN, DualRailMidSwapBellU, DualRailMidSwapBellN
+using LinearAlgebra
 using Test
 
-r_zalm = Register(2)
-initialize!(r_zalm[1:2], ZALMSpinPair(1e-3, 0.5, 0.5, 1, 1, 1, 1, 0.9, 1e-8, 1e-8, 1e-8, 0.99))
-@test ! iszero(observable(r_zalm[1:2], Z⊗Z))
+zalmU = ZALMSpinPairU(1e-3, 0.5, 0.5, 1, 1, 1, 1, 0.9, 1e-8, 1e-8, 1e-8, 0.99)
+zalmN = ZALMSpinPairN(1e-3, 0.5, 0.5, 1, 1, 1, 1, 0.9, 1e-8, 1e-8, 1e-8, 0.99)
+srmsU = SingleRailMidSwapBellU(0.9, 0.9, 0.5, 0.5, 1e-8, 0.99)
+srmsN = SingleRailMidSwapBellN(0.9, 0.9, 0.5, 0.5, 1e-8, 0.99)
+drmsU = DualRailMidSwapBellU(0.9, 0.9, 0.5, 0.5, 1e-8, 0.99)
+drmsN = DualRailMidSwapBellN(0.9, 0.9, 0.5, 0.5, 1e-8, 0.99)
 
-r_srms = Register(2)
-initialize!(r_srms[1:2], SingleRailMidSwapBell(0.9, 0.9, 0.5, 0.5, 1e-8, 0.99))
-@test ! iszero(observable(r_srms[1:2], Z⊗Z))
+r_zalmU = Register(2)
+initialize!(r_zalmU[1:2], zalmU)
+@test ! iszero(observable(r_zalmU[1:2], Z⊗Z))
 
-r_drms = Register(2)
-initialize!(r_drms[1:2], DualRailMidSwapBell(0.9, 0.9, 0.5, 0.5, 1e-8, 0.99))
-@test ! iszero(observable(r_drms[1:2], Z⊗Z))
\ No newline at end of file
+r_zalmN = Register(2)
+initialize!(r_zalmN[1:2], zalmN)
+@test ! iszero(observable(r_zalmN[1:2], Z⊗Z))
+
+r_srmsU = Register(2)
+initialize!(r_srmsU[1:2], srmsU)
+@test ! iszero(observable(r_srmsU[1:2], Z⊗Z))
+
+r_srmsN = Register(2)
+initialize!(r_srmsN[1:2], srmsN)
+@test ! iszero(observable(r_srmsN[1:2], Z⊗Z))
+
+r_drmsU = Register(2)
+initialize!(r_drmsU[1:2], drmsU)
+@test ! iszero(observable(r_drmsU[1:2], Z⊗Z))
+
+r_drmsN = Register(2)
+initialize!(r_drmsN[1:2], drmsN)
+@test ! iszero(observable(r_drmsN[1:2], Z⊗Z))
+
+@test tr(zalmU) < 1.0
+@test tr(express(zalmN).data) ≈ 1
```

Comment:
on the other hand, these should be exactly equal to 1 thanks to a special dispatch rule (so that `express` does not need to be used, nor `.data`)

Reply:
Wouldn't we still need to use express to convert it to `QuantumOptics` formalism so that [this](https://github.com/qojulia/QuantumOpticsBase.jl/blob/d2e4c8e98f3b7c0b38ead6434e9ffb0c5af1aaa4/src/operators_dense.jl#L186) specialised method can be called

Reply:
I would prefer if we do not rely on quantumoptics.jl at all for that.

In symbolic CAS tools usually 1 and 0 are treated more specially than other numbers. For the normalized density matrices here it is guaranteed that the trace is 1, so I would like to have a special-cased method for them that completely shortcuts any expression functionality.

`@test  tr(rho) = tr(express(rho))` should work for them.

Please add these special-cased methods for `tr` and I should be able to merge.

Reply:
@Abhishek-1Bhatt , bump